### PR TITLE
javaprocessor related updates

### PR DESCRIPTION
**Pull request by askmish**
- Fixed the javaprocessor build script
-- Changed git clone ops to clone only a single branch i.e. 1-0 for sawtooth-core
-- Removed sawtooth-core cloned directory after mvn install i.e. sawtooth-java-sdk is installed in the javaprocessor docker container

- Fixed the default ports for validator container in default docker-compose.yaml file

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -66,7 +66,7 @@ services:
     expose:
       - 4004
     ports:
-      - '4040:4004'
+      - '4004:4004'
```
Comment (danintel) [inline docker-compose.yaml:69]: I saw this. I thought it was mapped from TCP port 4004 to 4040 on purpose.  But I guess it was a mistake. If you are going to change this, you should change it for all:
```
$ grep 4040 *yaml
docker-compose.yaml:      - '4040:4004'
simplewallet-build-client-js.yaml:      - '4040:4004'
simplewallet-build-tp-cxx.yaml:      - '4040:4004'
simplewallet-build-tp-java.yaml:      - '4040:4004'
simplewallet-build-tp-js.yaml:      - '4040:4004'
simplewallet-build-tp-py.yaml:      - '4040:4004'
```

**Review by danintel — CHANGES_REQUESTED** [review]
See comment about 4040.

As you probably know. it looks like each SDK is about to be moved to their own repo, which will require a change in the future here and elsewhere:
https://github.com/search?q=org%3Ahyperledger+sawtooth+sdk

**Comment by askmish** [comment]
@danintel yep, we'll maintain a separate branch for sawtooth 1.0 and move the master to build on 1.1, once its available.

File: docker-compose.yaml (hunk 1 above)
Comment (askmish) [reply docker-compose.yaml:69]: Yep, that was the intention initially. I've been getting queries as to understand where did the port 4040 come from. :)  Hence, this update.

The changes for rest of the yaml files are scheduled for another PR, as they are not relevant to this PR's purpose.

**Review by danintel — APPROVED** [review]

**Pull request merged**